tutorials/shaders/compute_shaders.rst

```diff
@@ -377,3 +377,81 @@ shaders.
    GPU separately, which lets you compare how a similar algorithm can be
    implemented in two different ways (with the GPU implementation being faster
    in most cases).
+
+Includes Database
+-----------------
+.. warning::
+
+    The feature is experimental.
+
+Godot provides some shader includes through ``ShaderIncludeDB`` class. Include
+operation is done automatically by Godot. Shader includes are **not** files in
+our system.
+
+To see what headers are available, use
+``ShaderIncludeDB.list_built_in_include_files``. The contents of these files
+can be found in Godot's Git repository or by calling
+``ShaderIncludeDB.get_built_in_include_file``.
+
+The example below renders a transparent pulsating circle at the centre of the
+screen. The corners of the screen are coloured fully white. The edge of the
+circle is smoothed.
+
+.. code-block:: glsl
+
+  #[compute]
+  #version 450
+
+  #define MAX_VIEWS 2
+  #include "godot/scene_data_inc.glsl"
+
+  layout(local_size_x = 8, local_size_y = 8, local_size_z = 1) in;
+
+  layout(set = 0, binding = 0, std140) uniform SceneDataBlock {
+    SceneData data;
+    SceneData prev_data;
+  } scene_data_block;
+
+  layout(rgba16f, set = 0, binding = 1) uniform image2D screen;
+
+  void main(){
+    ivec2 uv = ivec2(gl_GlobalInvocationID.xy);
+
+    vec4 color = imageLoad(screen, uv);
+  
+    // Check UV is in the view.
+    if (uv.x >= scene_data_block.data.viewport_size.x || uv.y >= scene_data_block.data.viewport_size.y) {
+      return;
+    }
+  
+    float maxlen = max(scene_data_block.data.viewport_size.x, scene_data_block.data.viewport_size.y);
+  
+    vec2 center = vec2(scene_data_block.data.viewport_size.x / 2, scene_data_block.data.viewport_size.y / 2);
+    float r = length(abs(center - uv));
+    float alpha = smoothstep(maxlen * (0.55 + 0.3 * sin(scene_data_block.data.time)), maxlen * (0.2 + 0.2 * sin(scene_data_block.data.time)), r);
+  
+    // Calculate blend between full white and image's current color.
+    color = (1 - alpha) * vec4(1, 1, 1, 1) + alpha * color;
+    imageStore(screen, uv, color);
+  }
```

Comment:
The indentation here needs to be fixed to match the rest of the document, four spaces